tests/unit/restapi/v1/test_experiments.py

```diff
@@ -0,0 +1,859 @@
+# This Software (Dioptra) is being made available as a public service by the
+# National Institute of Standards and Technology (NIST), an Agency of the United
+# States Department of Commerce. This software was developed in part by employees of
+# NIST and in part by NIST contractors. Copyright in portions of this software that
+# were developed by NIST contractors has been licensed or assigned to NIST. Pursuant
+# to Title 17 United States Code Section 105, works of NIST employees are not
+# subject to copyright protection in the United States. However, NIST may hold
+# international copyright in software created by its employees and domestic
+# copyright (or licensing rights) in portions of software that were assigned or
+# licensed to NIST. To the extent that NIST holds copyright in this software, it is
+# being made available under the Creative Commons Attribution 4.0 International
+# license (CC BY 4.0). The disclaimers of the CC BY 4.0 license apply to all parts
+# of the software developed or licensed by NIST.
+#
+# ACCESS THE FULL CC BY 4.0 LICENSE HERE:
+# https://creativecommons.org/licenses/by/4.0/legalcode
+
+"""Test suite for experiment operations.
+
+This module contains a set of tests that validate the supported CRUD operations and
+additional functionalities for the experiment entity. The tests ensure that the
+experiments can be registered, retrieved, and deleted as expected through the REST API.
+"""
+
+from __future__ import annotations
+
+import datetime
+from typing import Any
+
+import pytest
+from flask.testing import FlaskClient
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.test import TestResponse
+
+from dioptra.restapi.routes import V1_AUTH_ROUTE, V1_EXPERIMENTS_ROUTE, V1_ROOT
+
+from ..lib import actions, helpers
+
+# -- Actions ---------------------------------------------------------------------------
+
+
+def modify_experiment(
+    client: FlaskClient,
+    experiment_id: int,
+    new_name: str | None = None,
+    new_entrypoints: list[int] | None = None,
+) -> TestResponse:
+    """Modify an experiment using the API.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to modify.
+        new_name: The new name to assign to the experiment.
+        new_entrypoints: The new entrypoints to associate with the experiment.
+
+    Returns:
+        The response from the API.
+    """
+
+    payload = {}
+
+    response = client.get(
```

Comment:
The actions should only have one call to the API. Since we already have access to the contents of the registered experiments in the tests via fixtures, any logic for choosing which fields to change can be placed there.

See https://github.com/usnistgov/dioptra/blob/db41860d97a22547d6decdf10232b6a2e2007617/tests/unit/restapi/v1/test_queue.py#L38 for example modify action.